server.js

```
@@ -0,0 +1,180 @@
+      });
+    }
+
+    // POST Request
+    if (req.method === "POST") {
+      if (req.url === "/elements") {
+        let q = querystring.parse(body);
+        let elName = q.elementName;
+        let elSymbol = q.elementSymbol;
+        let elNum = q.elementAtomicNumber;
+        let elDescript = q.elementDescription;
+
+        fs.writeFile(
+          path.join(__dirname, "/public", `${elName}.html`),
+          post(elName, elSymbol, elNum, elDescript),
+          err => {
+            if (err) throw err;
+
+            fs.readdir(path.join(__dirname, "/public"), (err, files) => {
+              if (err) throw err;
+
+              let count = 0;
+              let listOfElements = "\n";
+              files.forEach(e => {
+                if (
+                  path.extname(e) === ".html" &&
+                  e !== "404.html" &&
+                  e !== "index.html"
+                ) {
+                  listOfElements += `<li>
+  <a href="/${e}">${e.slice(0, -5)}</a>
+</li>`;
+                  listOfElements += "\n";
+                  count++;
+                  return;
+                }
+
+                fs.writeFile(
+                  path.join(__dirname, "/public", "index.html"),
+                  newHtml(listOfElements, count),
+                  err => {
+                    if (err) throw err;
+                  }
+                );
+              });
+            });
+          }
+        );
+
+        res.writeHead(200, {
+          "Content-Type": "application/json"
+        });
+        res.write(JSON.stringify({ success: true }));
+        res.end();
+      } else {
+        res.writeHead(200, {
+          "Content-Type": "application/json"
+        });
+        res.write(JSON.stringify({ success: false }));
+        res.end();
+      }
+    }
+
+    // DELETE Request
+    if (req.method === "DELETE") {
+      let publicDir = [];
+
+      fs.readdir(path.join(__dirname, "/public"), (err, files) => {
+        if (err) throw err;
```

This should return an error to the client using either a 400/500 error